### PR TITLE
Cart mapping

### LINKED ISSUE #6: As a customer/visitor, I want to be able to: change the quantity of a product in my cart
#5 

- [ ] option to change the quantity

### PR DESCRIPTION
**Pull request by kate-grant**
closes #6 
optimized cart and order routes. added dynamic quantity options and selection.

### CODE REVIEW

File: client/components/Cart.js [hunk 1]
```diff
@@ -1,41 +1,46 @@
...
   render() {
-    const items = this.props.order || [];
+    const items = this.props.cartItems || [];
     return (
       <div id="cart">
         <h2>Ready to checkout?</h2>
         <div id="cart-container">
           <h3>Your Items</h3>
           <div id="item-list">
+            {console.log('BEFORE MAP', this.props.cartItems)}
```
Comment (ssshhheeeyyy) [inline client/components/Cart.js:40]: Don't forget to delete this! :)

File: client/components/Cart.js [hunk 2]
```diff
@@ -1,41 +1,46 @@
...
             {items.length > 0 ? (
               items.map((item) => {
+                console.log('ITEM', item);
```
Comment (ssshhheeeyyy) [inline client/components/Cart.js:43]: another one

File: client/components/Cart.js [hunk 3]
```diff
@@ -46,8 +51,21 @@ export class Cart extends Component {
                       <div id="item-price">{item.product.price}</div>
                     </div>
                     <form>
-                      <select name="quantity" id="quantity-select">
+                      <select
+                        name="quantity"
+                        id="quantity-select"
+                        onChange={(e) => {
+                          console.log('IN CHANGE', item);
```
Comment (ssshhheeeyyy) [inline client/components/Cart.js:58]: and here

File: client/redux/orders.js [hunk 4]
```diff
@@ -42,13 +66,32 @@ export const fetchSingleCart = (id) => {
   };
 };
 
-export const addItem = (item) => {
+export const addItem = (productId, orderId, qty) => {
   // note: item {orderId:, productId:, quantity:}
-  console.log('in item thunk', item);
   return async (dispatch) => {
     try {
+      const item = { productId, orderId, quantity: qty };
```
Comment (ssshhheeeyyy) [inline client/redux/orders.js:73]: Don't forget to delete the quantity key

Comment (kate-grant) [reply client/redux/orders.js:73]: this works as is--I would also have to change the qty arg of this thunk to quantity for your comment to work--is that what you would prefer?

**Review by ssshhheeeyyy — COMMENTED** [review]
Hey! I left you some comments to delete console.logs :)

**Comment by kate-grant** [comment]
hey see commit fdf8da

File: client/redux/orders.js (hunk 4 above)
Comment (ssshhheeeyyy) [reply client/redux/orders.js:73]: Oh no, if it works nevermind! Disregard my comment :)

File: client/components/Cart.js (hunk 2 above)
Comment (kate-grant) [reply client/components/Cart.js]: commit fdf8da

File: client/components/Cart.js (hunk 3 above)
Comment (kate-grant) [reply client/components/Cart.js]: commit fdf8da

**Review by ssshhheeeyyy — APPROVED** [review]

**Pull request merged**